src/System.Private.CoreLib/shared/System/Threading/Tasks/Task.cs

```diff
@@ -3263,23 +3264,25 @@ internal void FinishContinuations()
             }
 
             // Not a single; it must be a list.
-            List<object?> continuations = (List<object?>)continuationObject;
+            List<object?> continuationList = (List<object?>)continuationObject;
 
             //
             // Begin processing of continuation list
             //
 
             // Wait for any concurrent adds or removes to be retired
-            lock (continuations) { }
-            int continuationCount = continuations.Count;
+            lock (continuationList) { }
+
+            // Operate over Span to skip extra bounds check from using List
+            Span<object?> continuations = CollectionsMarshal.AsSpan(continuationList);
```

Comment:
Having more than one continuation is a rare path already. Is it really worth optimizing this? How many continuations the task needs to have to make this optimization measurable?

Reply:
Removed this instance; left in the swaps for List.Enumerable as the 2x`sizeof(IntPtr)` Span is smaller than the 3x`sizeof(IntPtr)` List.Enumerator and its `foreach` conversion to `for(i<span.Length)` is more efficient than the List's `MoveNext`+`Current` which additionally incurs version and additional bounds checks.

> Is it really worth optimizing this?

Probably not significant relative to the continuations that are then triggered from `.WaitAll`; however my main motivation is: https://github.com/dotnet/corefx/pull/41306#pullrequestreview-292815363 

> Also, presumably there are places we should be using this in corefx? I don't love adding API we can't/don't ourselves use.

Ideally; I'd find places where it has a significant impact, however I'm a little time constrained atm.

Reply:
These unsafe or dangerous constructs should not be used just because of we can. Each use should be justified by measurable performance improvement.